src/tides.f90

```diff
@@ -573,4 +547,358 @@ subroutine setTidesInitFlag(this)
+            doodson(18)              = temp_Doodson
+          end if
+        end if
+      end do
+      ich = closeFile(ich)
+      !update the first call status
+      first_call = .false.
+      !assign the right order of magnitude to harmonic coefficients
+      dC_p = dC_p * 1.d-11
+      dC_m = dC_m * 1.d-11
+      dS_p = dS_p * 1.d-11
+      dS_m = dS_m * 1.d-11
+    end if
+
+  end subroutine initFES2004
+
+!=========================================================================
+!
+!> @anchor      getFES2004_corrections
+!!
+!> @brief       Compute harmonic coefficients corrections produced by the
+!!              FES2004 ocean tides model
+!!
+!> @author      Andrea Turchi (ATU)
+!!
+!> @date        <ul>
+!!                <li> 25.08.22 (initial design) </li>
+!!              </ul>
+!!
+!!------------------------------------------------------------------------
+  subroutine getFES2004_corrections(this, path, time_mjd, lmax, reduction, dC, dS)
+
+    implicit none
+    class(Tides_class)                          :: this
+    character(len=*), intent(in)                :: path                                   ! path for the input file
+    real(dp), intent(in)                        :: time_mjd                               ! time in modified julian date
+    integer, intent(in)                         :: lmax                                   ! maximum degree of harmonic coefficients
+    type(Reduction_type), intent(inout)         :: reduction
+    real(dp), dimension(5)                      :: F_vect                                 ! Delaunay arguments vector
+    real(dp)                                    :: theta_g                                ! Greenwich mean sidereal time in radians
+    real(dp)                                    :: theta_f                                ! argument of the tide constituent
+    real(dp)                                    :: m1, m2                                 ! auxiliaries to account for pole tide
+    real(dp)                                    :: fac                                    ! unnormalization factor
+    real(dp), dimension(1:18,2:6,0:6), save     :: dC_p                                   ! 3-D array of the prograde C coefficients of all tides
+    real(dp), dimension(1:18,2:6,0:6), save     :: dS_p                                   ! 3-D array of the prograde S coefficients of all tides
+    real(dp), dimension(1:18,2:6,0:6), save     :: dC_m                                   ! 3-D array of the retrograde C coefficients of all tides
+    real(dp), dimension(1:18,2:6,0:6), save     :: dS_m                                   ! 3-D array of the retrograde C coefficients of all tides
+    logical, save                               :: first_call = .true.                    ! initialization of the first call status
+    real, dimension(1:18), save                 :: doodson                                ! vector of the Doodson numbers of all tides
```

Comment:
This save is always dangerous. Is it really necessary? How often is this array used and maybe can it be moved to the module instead?